modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -172,6 +174,27 @@ public void onFailure(Exception e) {
                                     return;
                                 }
                             }
+                            String body;
+                            try {
+                                body = EntityUtils.toString(re.getResponse().getEntity());
+                            } catch (IOException ioe) {
+                                ioe.addSuppressed(e);
+                                fail.accept(ioe);
```

Comment:
I've reworked it so the IOException is the suppressed thing. I'll push in a bit.